experiments-core/src/foedus/ycsb/ycsb_load.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2014-2015, Hewlett-Packard Development Company, LP.
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details. You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ *
+ * HP designates this particular file as subject to the "Classpath" exception
+ * as provided by HP in the LICENSE.txt file that accompanied this code.
+ */
+
+#include <fcntl.h>
+#include <time.h>
+#include <unistd.h>
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+#include <sys/wait.h>
+
+#include <algorithm>
+#include <iostream>
+#include <string>
+#include <thread>
+#include <utility>
+#include <vector>
+
+#include "foedus/engine.hpp"
+#include "foedus/engine_options.hpp"
+#include "foedus/error_stack.hpp"
+#include "foedus/debugging/debugging_supports.hpp"
+#include "foedus/debugging/stop_watch.hpp"
+#include "foedus/fs/filesystem.hpp"
+#include "foedus/log/log_manager.hpp"
+#include "foedus/memory/engine_memory.hpp"
+#include "foedus/snapshot/snapshot_manager.hpp"
+#include "foedus/soc/shared_memory_repo.hpp"
+#include "foedus/soc/soc_manager.hpp"
+#include "foedus/storage/hash/hash_metadata.hpp"
+#include "foedus/storage/hash/hash_storage.hpp"
+#include "foedus/storage/masstree/masstree_cursor.hpp"
+#include "foedus/storage/masstree/masstree_metadata.hpp"
+#include "foedus/storage/masstree/masstree_page_impl.hpp"
+#include "foedus/storage/masstree/masstree_storage.hpp"
+#include "foedus/thread/thread.hpp"
+#include "foedus/xct/xct.hpp"
+#include "foedus/xct/xct_manager.hpp"
+#include "foedus/ycsb/ycsb.hpp"
+
+namespace foedus {
+namespace ycsb {
+
+ErrorStack ycsb_load_task(const proc::ProcArguments& args) {
+  thread::Thread* context = args.context_;
+  if (args.output_buffer_size_ < sizeof(std::pair<uint32_t, uint32_t>)) {
+    return ERROR_STACK(kErrorCodeUserDefined);
+  }
+  *args.output_used_ = sizeof(std::pair<uint32_t, uint32_t>);
```

Comment:
hm, let's define a struct KeyRange instead of std::pair so that it's clear what first/second means.